src/main/java/org/opentripplanner/middleware/tripmonitor/TrustedCompanion.java

```diff
@@ -0,0 +1,238 @@
+package org.opentripplanner.middleware.tripmonitor;
+
+import com.mongodb.client.model.Filters;
+import org.apache.logging.log4j.util.Strings;
+import org.opentripplanner.middleware.OtpMiddlewareMain;
+import org.opentripplanner.middleware.i18n.Message;
+import org.opentripplanner.middleware.models.OtpUser;
+import org.opentripplanner.middleware.models.RelatedUser;
+import org.opentripplanner.middleware.persistence.Persistence;
+import org.opentripplanner.middleware.utils.ConfigUtils;
+import org.opentripplanner.middleware.utils.HttpUtils;
+import org.opentripplanner.middleware.utils.I18nUtils;
+import org.opentripplanner.middleware.utils.NotificationUtils;
+import spark.Request;
+import spark.Response;
+
+import java.net.URLEncoder;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+import static com.mongodb.client.model.Filters.eq;
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static org.opentripplanner.middleware.tripmonitor.jobs.CheckMonitoredTrip.SETTINGS_PATH;
+import static org.opentripplanner.middleware.utils.I18nUtils.getLocaleFromString;
+import static org.opentripplanner.middleware.utils.I18nUtils.label;
+
+public class TrustedCompanion {
+
+    private TrustedCompanion() {
+        throw new IllegalStateException("Utility class");
+    }
+
+    private static final String AWS_API_SERVER = ConfigUtils.getConfigPropertyAsText("AWS_API_SERVER");
+    private static final String AWS_API_STAGE = ConfigUtils.getConfigPropertyAsText("AWS_API_STAGE");
+    private static final String OTP_UI_URL = ConfigUtils.getConfigPropertyAsText("OTP_UI_URL");
+    private static final String TRUSTED_COMPANION_CONFIRMATION_PAGE_URL =
+        ConfigUtils.getConfigPropertyAsText("TRUSTED_COMPANION_CONFIRMATION_PAGE_URL");
+    public static final String ACCEPT_KEY = "acceptKey";
+    public static final String USER_LOCALE = "userLocale";
+    public static final String EMAIL_FIELD_NAME = "email";
+
+    /** Note: This path is excluded from security checks, see {@link OtpMiddlewareMain#initializeHttpEndpoints()}. */
+    public static final String ACCEPT_DEPENDENT_PATH = "api/secure/user/acceptdependent";
+
+    /**
+     * Accept a request from another user to be their dependent. This will include both companions and observers. If
+     * successful redirect the user to the confirmation page, else redirect to the error page with related error.
+     */
+    public static OtpUser acceptDependent(Request request, Response response) {
+        Locale locale = getUserLocaleFromRequest(request);
```

Comment:
Get the locale using `I18nUtils.getOtpUserLocale(relatedUser)` instead.

Reply:
Unless I've missed something that is not possible hence providing the locale in the request.

Reply:
`relatedUser` is retrieved below, isn't it?

Reply:
Yes it is, but if we get that far, we don't need the locale.